### PR TITLE
REDACTED and LOBI staker

### PR DESCRIPTION
**Pull request by JExtor**

### CODE REVIEW

File: contracts/allocators/StakingManager.sol [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+import "../types/OlympusAccessControlled.sol";
+
+interface IStaking {
+    function stake( uint _amount, address _recipient ) external returns ( bool );
+    function unstake( uint _amount, bool _trigger ) external;
+    function claim ( address _recipient ) external;
+}
+
+contract OlympusStakingManager is OlympusAccessControlled {
```
Comment (ind-igo) [inline contracts/allocators/StakingManager.sol:17]: Should prob rename to something more specific. The name sounds general but this contract is specifically for Lobis and Redacted. Maybe `MetaGovernanceAllocator` ?

File: contracts/allocators/StakingManager.sol [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        require(address(authority.vault()) != address(treasury), "No change");
+        treasury = ITreasury(authority.vault());
+    }
+
+
+    /**
+     * @notice stakes either BTRFLY or LOBI from treasury
+     */
+    function stake(bool _redacted, uint _amount) external onlyGuardian {
+    (address staking, address token,) = _redactedOrLobi(_redacted);
```
Comment (ind-igo) [inline contracts/allocators/StakingManager.sol:45]: indentation is off

**Review by ind-igo — CHANGES_REQUESTED** [review]

File: contracts/allocators/StakingManager.sol [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        IStaking(staking).unstake(_amount, false);
+
+        // send token back to treasury
+        IERC20(token).safeTransfer(address(treasury), _amount);
+    }
+
+    /**
+     * @notice returns addresses depending on wanting to interact with redacted or lobi
+     */
+    function _redactedOrLobi(bool _redacted) internal view returns (address staking, address token, address stakedToken) {
```
Comment (lolchocotaco) [inline contracts/allocators/StakingManager.sol:82]: I think this is okay for now. If we end up in a situation where we need another allocator for the same purpose, we should consolidate to a more generic implementation.

**Review by lolchocotaco — COMMENTED** [review]
I think this is okay for now. If we end up in a situation where we need another allocator for the same purpose, we should consolidate to a more generic implementation.

File: contracts/allocators/StakingManager.sol (hunk 1 above)
Comment (JExtor) [reply contracts/allocators/StakingManager.sol]: done

File: contracts/allocators/StakingManager.sol (hunk 2 above)
Comment (JExtor) [reply contracts/allocators/StakingManager.sol]: done

File: contracts/allocators/StakingManager.sol (hunk 3 above)
Comment (JExtor) [reply contracts/allocators/StakingManager.sol]: Yea, was thinking the same thing, my thinking was that the big two we were wanting for BTRFLY and LOBI, wanted to make it where there was no address being passed in and only known addresses could be staked to with this implementation. Can always go back and deploy new contract if we come to situation where we are staking to other Olympus style protocols besides these two.

File: test/tokens/sOhm.test.ts [hunk 4]
```diff
@@ -158,8 +159,11 @@ describe("sOhm", () => {
       });
     });
 
-    describe("circulatingSupply", () => {
+    // FIXME
```
Comment (lolchocotaco) [inline test/tokens/sOhm.test.ts:162]: What happened here?  Who gon fix. 
can also do `describe.skip`.

**Pull request merged**